### PR TITLE
Added Password files and optimzed

### PR DESCRIPTION
**Pull request by miguelgargallo**
### Description

I added the Password files and removed the parts that are unnecessary according the new Anse version

### Linked Issues

#5 Password implementation  

### Additional context

I just added the files and optimized, this project is awesome and I invest my time to improve it!

### CODE REVIEW

**Comment by ddiu8081** [comment]
I'm glad you're contributing to this feature, however it's not perfect yet.

For one thing, it lacks api-level authentication - Anse does not use `api/generate` route, which means that in Anse, there's no verification when sending a prompt, and users can easily bypass the password authentication page by blocking the request from the index page.

Also, I don't think it works in Docker environments. `chatgpt-demo` has been a problem in Docker, it doesn't read environment variables very well. I don't plan to use environment variables until that's fixed.

Still, thanks for the commit, and you can go ahead and finish your requirements in the fork version!

File: src/pages/password.astro [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+import Layout from '../layouts/Layout.astro'
+---
+
+<Layout title="Password Protection">
```
Comment (Sysamin478) [inline src/pages/password.astro:5]: Protect

File: src/pages/index.astro [hunk 2]
```diff
@@ -21,3 +21,23 @@ import BuildStores from '@/components/client-only/BuildStores'
...
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        pass: password,
+      }),
+    })
+    const responseJson = await response.json()
+    if (responseJson.code !== 0)
+      window.location.href = '/password'
```
Comment (Sysamin478) [inline src/pages/index.astro:40]: Protect

File: src/pages/password.astro [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  <main class="h-screen col-fcc">
+    <div class="op-30">Please input password</div>
+    <div id="input_container" class="flex mt-4">
+      <input id="password_input" type="password" class="gpt-password-input" />
```
Comment (Sysamin478) [inline src/pages/password.astro:9]: Protect

**Pull request closed without merging**